### PR TITLE
Allow PSF components to be in folder other than package root

### PR DESCRIPTION
**Pull request by TimMangan**
Previously, the config.json, PsfRuntime, and fixup dlls needed to be located at the root folder of the package.  This PR allows them to be located in a sub-folder along with the exe files.

This simplifies the repackaging process, while not affecting packages where the components are located in the root folder.

### CODE REVIEW

**Comment by TimMangan** [comment]
NOTE: There are only two files that should change in this PR, config.cpp and CreateProcessHook.cpp

**Comment by dhoehna** [comment]
Hey Tim.  I got the tests working after your PR was accepted.  Can you please re-base the develop branch into your master branch and run the tests to make sure they run?  I can provide guidance on how to fix the code if the tests aren't green.

Here is how you run the tests

1. Open an admin VS developer command prompt.
2. Navigate to your repo
3. Run "buildall.cmd"
4. When that finished open an admin power shell prompt.
5. Make sure the execution policy for scripts is set to either "remoteSigned" or "Unrestricted" you can check the execution policy by running the cmdlet "get-executionpolicy"
6. Head to the tests folder and run "runtests.ps1"
7. Wait for the tests to finish.

If 0 tests failed, awesome.  If not we need to see why they failed and how to fix them.

**Comment by TimMangan** [comment]
At ignite this week, so I'll check it when back next week.
FYI: Added a new small PR to complete the ability to put the PSF components in folders other than the root. 2 files in psfruntime and shouldn't affect existing tests but I'll run them.

Tim

Get Outlook for Android<https://aka.ms/ghei36>

________________________________

**Comment by dhoehna** [comment]
Tim.  The reason why there are so many files is because the PR is going into Master.   Can you change the PR target to develop please?

**Comment by TimMangan** [comment]
Yes.

Get Outlook for Android<https://aka.ms/ghei36>

________________________________

**Comment by TimMangan** [comment]
Ok. Retargeted to the develop branch.

File: PsfRuntime/CreateProcessHook.cpp [hunk 1]
```diff
@@ -174,7 +195,11 @@ BOOL WINAPI CreateProcessFixup(
             }
         }
     }
+<<<<<<< HEAD
```
Comment (dhoehna) [inline PsfRuntime/CreateProcessHook.cpp]: Fix the merge conflict text please.

**Comment by dhoehna** [comment]
Hey Tim.  Thank you for moving the PR over.  I don't think your rebase worked.  I don't see files that I added in my PR.

Here is the PR and all the files that were changed and/or added.  https://github.com/microsoft/MSIX-PackageSupportFramework/commit/ec885566032716e7e9da9e3af7569be64a5444b7

Without these changes the tests on your branch are failing.  The rebase should fix the tests.

**Comment by TimMangan** [comment]
Will look at it Friday.

Get Outlook for Android<https://aka.ms/ghei36>

________________________________

**Comment by TimMangan** [comment]
Darren,

I will confess to being baffled by GitHub. I've pulled your dev branch, built and ran the tests successfully.  How to rebase my fork is a mystery to me.  I try what I think I should do in VS Code, but never seem to get your changes.

I'm tempted to just create a new fork and add my changes to that, close the PR and create a new one.  If you have advice, I'd love to hear it.

PS: Your instructions for building and running the tests were great, but perhaps add a note for the dependency on the older version of the SDK.

**Comment by dhoehna** [comment]
Hey Tim,

No worries.  git is a confusing beast.  Thank you for trying.

Here are instructions on how to rebase the fork into your current branch.  Here is where I am getting the instructions.

https://gist.github.com/ravibhure/a7e0918ff4937c9ea1c456698dcd58aa

1. in a cmd run "git remote add upstream https://github.com/microsoft/MSIX-PackageSupportFramework.git"
2. run "git fetch upstream"

I tried this rebase on my computer and there are conflicts between what is on develop and what your master branch has.

My suggestion for dealing with this is to override your local copy with everything from develop and manually re-add all your changes.  You can do something like this.

1. Copy each file you changed to a different location outside of your repository.
2. Run "git rebase upstream/develop -X --theirs" 
3. Fix any merge conflicts (The only one I found was in CreateProcessHook.cpp
4. Reapply your changes manually.

If you are still having problems let me know and I can do the rebasing on my side and push the changes to your master branch.  From there you can re-apply your changes.

Good idea on the SDK documentation idea.

**Comment by TimMangan** [comment]
I have tried, but don’t seem to be able to get my fork to sync in changes made to the Microsoft repository.

I did add a commit (ecbe04039bb5de8ddb68100e6011e941a0588fc1) to my develop branch that has the two files associated with this PR.  I have also downloaded the Microsoft develop branch, added the two files and ran the tests successfully.

If you can manage to get that commit to work for you great.  Otherwise I feel I’ll need to remove the PR, delete the fork, create a new one, and apply the two files and recreate the PR.

Let me know what you want me to do.

Tim

**Comment by dhoehna** [comment]
Hey Tim,

Thank you for trying.  No worries about not getting your branch to sync.  But, it will be something that has to happen when there are other PR's to develop.

If you only changed two files the best thing to do would be to make your fork again from develop.  You can then re-apply your changes and make a fresh PR.

When you have your other PR up I'll take a look at it.

Sound good?

**Comment by TimMangan** [comment]
Closing this PR to create a replacement.

**Comment by TimMangan** [comment]
Closing this PR to create a replacement.

**Comment by TimMangan** [comment]
PR #97 was closed, the fork removed and rebuilt.  A new P# #99 replaces it.

**Pull request closed without merging**